Readme.md

```diff
@@ -477,50 +477,152 @@ setTimeout(function() {
 }, 1000);
 ```
 
-## Use slashes for comments
+## Writing Comments
 
-Use slashes for both single line and multi line comments. Try to write
-comments that explain higher level mechanisms or clarify difficult
-segments of your code. Don't use comments to restate trivial things.
+*Referenced from [Google Style Guide][googlecomments]:*
+
+All files, classes, methods and properties should be documented with
+[JSDoc][jsdoc] comments with the appropriate annotations, tags and types.
+Textual descriptions for properties, methods, method parameters and method
+return values should be included unless obvious from the property, method, or
+parameter name.
+
+Complete sentences are recommended but not required. Complete sentences should
+use appropriate capitalization and punctuation.
 
 *Right:*
 
 ```js
-// 'ID_SOMETHING=VALUE' -> ['ID_SOMETHING=VALUE', 'SOMETHING', 'VALUE']
-var matches = item.match(/ID_([^\n]+)=([^\n]+)/));
+/**
+ * A JSDoc comment should begin with a slash and 2 asterisks.
+ * Inline tags should be enclosed in braces like {@code this}.
+ * @desc Block tags should always start on their own line.
+ */
 
-// This function has a nasty side effect where a failure to increment a
-// redis counter used for statistics will cause an exception. This needs
-// to be fixed in a later iteration.
-function loadUser(id, cb) {
-  // ...
-}
+/** One line block comments like this are okay. */
+```
 
-var isSessionValid = (session.expires < Date.now());
-if (isSessionValid) {
-  // ...
-}
+*Wrong:*
+
+```js
+/*
+Multi-line block comments should begin with two asterisks.
+ */
+
+/* C-style block comments are discouraged. */
+```
+
+If a line break is needed in block comments, you should indent the subsequent
+lines with 4 spaces indent. The description paragraph in the block comments does
```

Comment:
I think we should try to stick to 2 spaces everywhere. People are bound to get confused about how many spaces they should use where.